tests/test_template.py

```diff
@@ -0,0 +1,135 @@
+                                            [ 3.55439469e-04,  3.89820231e-02, -3.35281297e-03,  7.93009278e-04, 8.30347900e-02, -3.45150712e-01],
+                                            [ 5.44893709e-03,  9.30878023e-03, -5.33374090e-01,  6.13889584e-02,-1.16931386e-01,  3.45150712e-01],
+                                            [ 3.81099656e-04, -4.79703149e-02,  5.78754312e-01,  0.00000000e+00, 3.38965962e-02,  0.00000000e+00]]
+                                  ), delta=self.tol)
+
+    def setUp(self):
+        super().setUp()
+        np.random.seed(8)
+        self.num_wires = 4
+
+    def test_cvqnn(self):
+
+        dev = qml.device('default.gaussian', wires=self.num_wires)
+
+        weights = [[
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #transmittivity angles
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #phase angles
+            np.random.uniform(0.1, 0.7, self.num_wires), #squeezing amounts
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #transmittivity angles
+            np.random.uniform(0, 2*np.pi, int(self.num_wires*(self.num_wires-1)/2)), #phase angles
+            np.random.uniform(0.1, 2.0, self.num_wires), #displacement magnitudes
+            np.random.uniform(0.1, 0.3, self.num_wires) #kerr parameters
+        ] for l in range(2)]
+
+        with patch.object(Kerr, '__init__', return_value=None) as _: #Kerr() does not work on any core device, so we have to mock it here with a trivial class
+            @qml.qnode(dev)
+            def circuit(weights):
+                qml.template.CVNeuralNet(weights, wires=range(self.num_wires))
+                return qml.expval.X(wires=0)
+
+            circuit(weights)
+
+class TestVariationalClassifiyer(BaseTest):
+    """Tests for the CircuitCentric from the pennylane.template module."""
+
+    def setUp(self):
+        super().setUp()
+        np.random.seed(0)
+
+    def test_variational_classifyer(self):
+        """Tests the CircuitCentric for various parameters."""
+        outcomes = []
+        for num_wires in range(2,4):
+            for num_layers in range(1,3):
+
+                dev = qml.device('default.qubit', wires=num_wires)
+
+                @qml.qnode(dev)
+                def circuit(weights, x=None):
+                    qml.BasisState(x, wires=range(num_wires))
+                    qml.template.CircuitCentric(weights, True, wires=range(num_wires))
+                    return qml.expval.PauliZ(0)
+
+                weights=np.random.randn(num_layers, num_wires, 3)
+                outcomes.append(circuit(weights, x=np.array(np.random.randint(0,1,num_wires))))
+        self.assertAllAlmostEqual(np.array(outcomes), np.array([-0.29242496,  0.22129055,  0.07540091, -0.77626557]), delta=self.tol)
```

Comment:
Similar question to above: how did you determine these numbers?

Reply:
Also here, just an execution test.